src/export/ProfileExporter.ts

```diff
@@ -5,27 +5,22 @@ import { FHIRDefinitions } from '../fhirdefs';
 import { logger } from '../utils/FSHLogger';
 
 export class ProfileExporter extends StructureDefinitionExporter {
-  constructor(FHIRDefs: FHIRDefinitions) {
-    super(FHIRDefs);
+  constructor(FHIRDefs: FHIRDefinitions, tank: FSHTank) {
+    super(FHIRDefs, tank);
   }
 
   /**
    * Exports Profiles to StructureDefinitions
-   * @param {FSHTank} tank - The FSH tank we are exporting
    * @returns {StructureDefinition[]}
    */
-  export(tank: FSHTank): StructureDefinition[] {
-    const structDefs: StructureDefinition[] = [];
-    for (const doc of tank.docs) {
-      for (const profile of doc.profiles.values()) {
-        try {
-          const structDef = this.exportStructDef(profile, tank);
-          structDefs.push(structDef);
-        } catch (e) {
-          logger.error(e.message);
-        }
+  export(): StructureDefinition[] {
+    for (const profile of this.tank.getAllProfiles()) {
+      try {
+        this.exportStructDef(profile);
+      } catch (e) {
+        logger.error(e.message);
       }
```

Comment:
For both `ProfileExporter` and `ExtensionExporter`, I think looping in this way can be inefficient. Let's say we have Foo with parent Bar with parent Baz. If Foo is exported first, then Bar and Baz (by my understanding) will be exported during the export of Foo. But then on the next iteration of the loop won't we re-export Bar, getting the Baz SD, copying it, and applying all the rules again? Then the next iteration I think we will re-export Baz, getting the Resource SD, copying it, and applying all the rules. I have stepped through the code and I'm pretty confident this is what is happening. I guess it isn't explicitly going to cause incorrect behavior, but I think it could be much more efficient if when we are exporting a SD, we check if that SD has already been exported, either here or at the beginning of the `exportStructDef` function.

Reply:
Good call. I moved the check for whether something was exported to the beginning of `exportStructDef` rather than the end.